index.d.ts

```diff
@@ -8,15 +8,10 @@
 //   Michal Adamczyk <https://github.com/mradamczyk>
 //   Marvin Hagemeister <https://github.com/marvinhagemeister>
 
-export type ClassValue = string | number | ClassDictionary | ClassArray | undefined | null | boolean;
+export type Value = string | number | boolean | undefined | null;
+export type Mapping = { [key: string]: Value };
+export type Argument = Value | Mapping | Argument[];
 
-export interface ClassDictionary
-{
-	[id: string]: boolean | undefined | null;
-}
-
-export interface ClassArray extends Array<ClassValue> {}
-
-export function classNames(...args: ClassArray): string;
```

Comment:
Removed the `Class` prefix (redundant when you are importing).

The mapping of `string` to truthy values was different between between arrays and objects,  which is not what our tests dictate.   Changed that.